### ISSUE TITLE
Applying plugin changes project repositories

### ISSUE THREAD

**Issue by stempler** [bmuschko/gradle-docker-plugin#257]
I'm having some trouble with the changes introduced in [this pull request](https://github.com/bmuschko/gradle-docker-plugin/pull/218).

In my Groovy project where I use the docker plugin this leads to the Gradle plugins Maven repository (http://plugins.gradle.org/m2) being added before any repositories that I specify in my build script.

Simplified build script:

```groovy
plugins {
  id 'groovy'
  id 'application'
  id 'com.bmuschko.docker-java-application' version '3.0.1'
}

repositories {
  // this needs to be defined before jcenter/MavenCentral for retrieving JAI
  maven {
    url 'http://download.osgeo.org/webdav/geotools/'
  }
  jcenter()
}
...
```

Usually that would not be a problem, but for Java Advanced Imaging (`javax.media:jai_core:1.1.3`) the order of repositories checked is quite important. If I update the Docker plugin to 3.0.0/3.0.1 for any of my projects having this dependency, the builds will fail, because Gradle tries to retrieve the dependency from the Gradle plugin repository where the POM is available, but not the JAR.

```
* What went wrong:
Could not resolve all dependencies for configuration ':compileClasspath'.
> Could not find jai_core.jar (javax.media:jai_core:1.1.3).
  Searched in the following locations:
      https://plugins.gradle.org/m2/javax/media/jai_core/1.1.3/jai_core-1.1.3.jar
```

Gradle complains when I try to move the `repositories` section before the `plugins` section, so the changes in #218 will always add the buildscript repositories to the project's repositories, before I can add them in the build script.

Why does the plugin need to adapt the project's repositories at all? Shouldn't the plugin use the buildscript repositories and dependencies?
If it is really necessary, can it be made configurable? Or maybe made to apply at a later stage, when repositories might have been added in the build script? (e.g. with `project.afterEvaluate { ... }`)

**Comment by stempler** [comment]
If anyone else is having that problem, workaround is adding the repositories that should be checked first to the buildscript repositories:

```
buildscript {
  repositories {
    // these need to before jcenter/MavenCentral for retrieving JAI
    maven {
      url 'http://download.osgeo.org/webdav/geotools/'
    }
  }
}

plugins {
  id 'groovy'
  id 'application'
  id 'com.bmuschko.docker-java-application' version '3.0.1'
}

repositories {
  jcenter()
}
...
```

**Comment by orzeh** [comment]
@stempler this looks like a bug. As you suggest adding build script repositories should be done in  `project.afterEvaluate` (if needed). @cdancy can you fix that?

**Comment by cdancy** [comment]
@orzeh @stempler definitely a bug. I'll put in a fix for this today.

**Comment by cdancy** [comment]
@stempler just released version 3.0.2. Please gives things a go and let me know if you have any issues. Thanks.

**Comment by stempler** [comment]
@cdancy @orzeh Awesome, thanks a lot! Working for me w/o issues.
I left a small comment in the respective pull request, not sure if it is relevant.
